packages/device_info/lib/device_info.dart

```diff
@@ -0,0 +1,231 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
```

Comment:
Still doesn't feel right. You may _already be_ in a platform-specific context (i.e. there's no `defaultTargetPlatform` check in sight) when you are using the plugin. You'd want the IDE to assist you with inspecting the properties of the plugin API, but it returns either an opaque `DeviceInfo` with either no members 😮  or lacks platform-specific members, which requires that you go read the docs to learn that you need to cast the result to something else (or worse, you decide that the plugin is missing this feature and walk away).

With platform-specific getters, the the IDE can suggest {iosInfo, androidInfo} to pick from, once you pick one, say `androidInfo`, the IDE will continue assisting you with the available properties.

I still think composition is better than inheritance for this use-case.

I'll switch to class container-based API for consistency with other plugins. QQ: re mocking, if you can already mock out the `MethodChannel`, what's the value of also wrapping the API into a container class?